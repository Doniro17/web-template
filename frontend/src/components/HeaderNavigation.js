import React, { useEffect } from 'react';
import { Link, useHistory } from 'react-router-dom';
import { FaUserAlt } from 'react-icons/fa';
import { IoMdLogOut } from 'react-icons/io';
import API from '../utils/API';

export default function HeaderNavigation(props) {
  const {
    user: { roles, login },
    setUserData,
    setLoading,
  } = props;
  const history = useHistory();
  useEffect(() => {
    const getUserData = async () => {
      setLoading(true);
      const result = await API.get('users/profile', {
        withCredentials: true,
      }).catch((e) => {
        if (e.response.status === 401) {
          history.push('/login');
        }
      });
      setUserData(result.data.id, result.data.username, result.data.role);
      setLoading(false);
    };
    getUserData();
  }, []);
  async function onLogoutClick() {
    setUserData('', '', []);
    history.push('/login');
    await API.post('users/logout', {
      withCredentials: true,
    });
  }
  function NavigationBar() {
    return (
      <>
        {roles.includes('client') && (
          <>
            <Link className="links__item" to="/products">
              Product
            </Link>
            <Link className="links__item" to="/orders">
              Orders
            </Link>
          </>
        )}
        {roles.includes('courier') && (
          <>
            <Link className="links__item" to="/deliveries">
              Delivery
            </Link>
            <Link className="links__item" to="/orders?lala=lolo">
              Requests
            </Link>
          </>
        )}
        {login.length > 0 && (
          <>
            <div className="links__profile">
              <Link className="profile__name" to="/deliveries">
                {login}
              </Link>
              <FaUserAlt className="profile__icon" />
              <IoMdLogOut onClick={onLogoutClick} className="profile__logout" />
            </div>
          </>
        )}
      </>
    );
  }

  return (
    <header className="header">
      <nav className="header__links">
        <div className="links__item">Logo</div>
        <NavigationBar />
        {/* <div className="links__profile">
          <Link className="profile__name" to="/deliveries">
            {login}
          </Link>
          <FaUserAlt className="profile__icon" />
          <IoMdLogOut className="profile__logout" />
        </div> */}
      </nav>
    </header>
  );
}
