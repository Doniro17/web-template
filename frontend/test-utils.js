import React from 'react'; // eslint-disable-line no-unused-vars

import { initFontAwesomeLibrary } from './src/utils';

import '@testing-library/jest-dom/extend-expect';

initFontAwesomeLibrary();

// re-export everything
export * from '@testing-library/react';
