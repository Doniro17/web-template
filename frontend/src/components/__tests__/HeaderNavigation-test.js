import { render } from 'test-utils';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter } from 'react-router-dom';
import UserContext from '../../App/context';
import HeaderNavigation from '../HeaderNavigation';

let context = { user: { roles: ['client', 'courier'] } };

describe('HeaderNavigation', () => {
  it('renders correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <UserContext.Provider value={context}>
          <HeaderNavigation />
        </UserContext.Provider>
      </BrowserRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it('Correct navigation render for client', () => {
    context = { user: { roles: ['client'] } };
    const { queryByText } = render(
      <BrowserRouter>
        <UserContext.Provider value={context}>
          <HeaderNavigation />
        </UserContext.Provider>
      </BrowserRouter>,
    );
    expect(queryByText('Product')).toBeInTheDocument();
    expect(queryByText('Orders')).toBeInTheDocument();
    expect(queryByText('Delivery')).not.toBeInTheDocument();
    expect(queryByText('Requests')).not.toBeInTheDocument();
  });
  it('Correct navigation render for courier', () => {
    context = { user: { roles: ['courier'] } };
    const { queryByText } = render(
      <BrowserRouter>
        <UserContext.Provider value={context}>
          <HeaderNavigation />
        </UserContext.Provider>
      </BrowserRouter>,
    );
    expect(queryByText('Product')).not.toBeInTheDocument();
    expect(queryByText('Orders')).not.toBeInTheDocument();
    expect(queryByText('Delivery')).toBeInTheDocument();
    expect(queryByText('Requests')).toBeInTheDocument();
  });
});
