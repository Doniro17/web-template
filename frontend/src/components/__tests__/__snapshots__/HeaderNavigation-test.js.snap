// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNavigation renders correctly 1`] = `
<header
  class="header"
>
  <nav
    class="header__links"
  >
    <div
      class="links__item"
    >
      Logo
    </div>
    <a
      class="links__item"
      href="/products"
    >
      Product
    </a>
    <a
      class="links__item"
      href="/orders?status=createt"
    >
      Orders
    </a>
    <a
      class="links__item"
      href="/deliveries"
    >
      Delivery
    </a>
    <a
      class="links__item"
      href="/orders?lala=lolo"
    >
      Requests
    </a>
  </nav>
</header>
`;
